test/grouped_convnd_bwd_weight/test_grouped_convnd_bwd_weight_interface_xdl.cpp

```diff
@@ -94,86 +168,261 @@ class TestGroupedConvndBwdWeight : public ::testing::Test
+        }
     }
 };
 
-using GNHWC = ck::tensor_layout::convolution::GNHWC;
-using NHWGC = ck::tensor_layout::convolution::NHWGC;
+TEST_F(TestGroupedConvndBwdWeight, TestGroupedConvndBwdWeight_NDimSpatial_1_Filter_1x1)
+{
+    this->conv_param = {
+        1, 2, 4, 192, 192, {1, 1, 1}, {28, 28, 28}, {1, 1, 1}, {1, 1, 1}, {0, 0, 0}, {0, 0, 0}};
+    this->template Run<1>();
+}
 
-using GKYXC = ck::tensor_layout::convolution::GKYXC;
+TEST_F(TestGroupedConvndBwdWeight, TestGroupedConvndBwdWeight_NDimSpatial_1_Filter_3x3)
+{
+    this->conv_param = {
+        1, 2, 4, 192, 192, {3, 3, 3}, {28, 28, 28}, {1, 1, 1}, {1, 1, 1}, {0, 0, 0}, {0, 0, 0}};
+    this->template Run<1>();
+}
 
-using GNHWK = ck::tensor_layout::convolution::GNHWK;
-using NHWGK = ck::tensor_layout::convolution::NHWGK;
+TEST_F(TestGroupedConvndBwdWeight, TestGroupedConvndBwdWeight_NDimSpatial_2_Filter_1x1)
+{
+    this->conv_param = {
+        2, 2, 4, 192, 192, {1, 1, 1}, {28, 28, 28}, {1, 1, 1}, {1, 1, 1}, {0, 0, 0}, {0, 0, 0}};
+    this->template Run<2>();
+}
 
-using KernelTypes =
-    ::testing::Types<std::tuple<GNHWK, GKYXC, GNHWC>, std::tuple<NHWGK, GKYXC, NHWGC>>;
+TEST_F(TestGroupedConvndBwdWeight, TestGroupedConvndBwdWeight_NDimSpatial_2_Filter_3x3)
+{
+    this->conv_param = {
+        2, 2, 4, 192, 192, {3, 3, 3}, {28, 28, 28}, {1, 1, 1}, {1, 1, 1}, {0, 0, 0}, {0, 0, 0}};
+    this->template Run<2>();
+}
 
-template <typename Tuple>
-class TestGroupedConvndBwdWeightDefault
-    : public TestGroupedConvndBwdWeight<Tuple, ConvBwdWeightDefault>
+TEST_F(TestGroupedConvndBwdWeight, TestGroupedConvndBwdWeight_NDimSpatial_3_Filter_1x1)
 {
-};
+    this->conv_param = {
+        3, 2, 4, 192, 192, {1, 1, 1}, {28, 28, 28}, {1, 1, 1}, {1, 1, 1}, {0, 0, 0}, {0, 0, 0}};
+    this->template Run<3>();
+}
 
-template <typename Tuple>
-class TestGroupedConvndBwdWeightFilter1x1
-    : public TestGroupedConvndBwdWeight<Tuple, Filter1x1Stride1Pad0>
+TEST_F(TestGroupedConvndBwdWeight, TestGroupedConvndBwdWeight_NDimSpatial_3_Filter_3x3)
 {
-};
+    this->conv_param = {
+        3, 2, 4, 192, 192, {3, 3, 3}, {28, 28, 28}, {1, 1, 1}, {1, 1, 1}, {0, 0, 0}, {0, 0, 0}};
+    this->template Run<3>();
+}
 
-TYPED_TEST_SUITE(TestGroupedConvndBwdWeightDefault, KernelTypes);
-TYPED_TEST_SUITE(TestGroupedConvndBwdWeightFilter1x1, KernelTypes);
+TEST_F(TestGroupedConvndBwdWeight, TestGroupedConvndBwdWeight_NDimSpatial_1_Stride_1x1)
+{
+    this->conv_param = {
+        1, 2, 4, 192, 192, {1, 1, 1}, {28, 28, 28}, {1, 1, 1}, {1, 1, 1}, {0, 0, 0}, {0, 0, 0}};
+    this->template Run<1>();
+}
 
-TYPED_TEST(TestGroupedConvndBwdWeightFilter1x1, SpecializationCheck)
+TEST_F(TestGroupedConvndBwdWeight, TestGroupedConvndBwdWeight_NDimSpatial_1_Stride_2x2)
 {
-    // Check filter 3,3 instead of 1,1
-    this->conv_param  = {2, 2, 4, 192, 192, {3, 3}, {28, 28}, {1, 1}, {1, 1}, {0, 0}, {0, 0}};
-    bool is_supported = this->template Run<2>();
-    EXPECT_FALSE(is_supported);
+    this->conv_param = {
```

Comment:
I dont understand changes in this file. This test is to check instance factory interface. Not performance or correctness. We have second test_grouped_convnd_bwd_weight for correctness.